packages/devtools/lib/src/ui/ui_utils.dart

```diff
@@ -187,6 +208,36 @@ class ServiceExtensionButton {
   PButton button;
 
   void _click() {
+    switch (extensionDescription.extension) {
+      case 'ext.flutter.debugAllowBanner':
```

Comment:
Instead of adding a switch here, add the analytics name to the extensionDescription object and make it required.
That way we won't miss adding analytics for extensions.

Reply:
Done.

Reply:
I still see the switch. You should be able to remove the switch statement completely by adding the ga analytics tag to the extensionDescription object which already has a bunch of metadata describing a service extension.